radis/lbl/base.py

```diff
@@ -1968,14 +2071,116 @@ def calc_populations_eq(self, Tgas):
         # Calculate population
         # ----------------------------------------------------------------------
         # equilibrium: Boltzmann in any case
-        df1["nu"] = df1.gu.values * exp(-hc_k * df1.Eu.values / Tgas) / df1.Qgas.values
+
+        # multiple-isotopes in database
+
+        # extract Qgas in the required form from the dictionary
+        Qgas = self.getQ_from_dict(df1, Qgas_dict)
+
+        df1["nu"] = df1.gu.values * exp(-hc_k * df1.Eu.values / Tgas) / Qgas
 
         assert "nu" in self.df1
 
         self.profiler.stop("calc_eq_population", "Calculated equilibrium populations")
 
         return
 
+    def getQ_from_dict(self, df, Qgas):
+
+        """Returns the required value of Qgas for calculation of linestrength
+        from Qgas dictionary
+
+        Parameters
+        ----------
+        df: dataframe
+        Qgas: dictionary that contais Qgas values of various isotopes
+
+        Returns
+        -------
+        The required value of Qgas either in the form of just a float value or array-like value
+
+        """
+
+        # multiple isotopes
+        if "iso" in df:
+            Q_req = df["iso"].map(Qgas)
+
+        # single isotopes
+        else:
+            Qgas = list(Qgas.values())
+            Q_req = Qgas[0]
+
+        return Q_req
+
+    def get_Qgas(self, df, Tgas):
+
+        """Aggregates the values of Qgas
+
+        Parameters
+        ----------
+        df: dataframe
+        Tgas: float (K)
+            gas temperature
+
+        Returns
+        -------
+        Returns Qgas as a dictionary with isotope values as its keys
+
+        """
+
+        Qgas = {}
+
+        if "id" in df and "iso" in df:
+
+            dgb = df.groupby(by=["id", "iso"])
+
+            for (id, iso), idx in dgb.indices.items():
+                molecule = get_molecule(id)
+                state = self.input.state
+                parsum = self.get_partition_function_interpolator(molecule, iso, state)
+                Qgas[iso] = parsum.at(Tgas)
+
+                if radis.DEBUG_MODE:
+                    assert (df.loc[idx, "id"] == id).all()
+                    assert (df.loc[idx, "iso"] == iso).all()
+
+        elif "id" not in df and "iso" in df:
+
+            dgb = df.groupby(by=["id"])
+
+            id = df.attrs["id"]
+            iso1 = list(dgb.indices.keys())[0]
```

Comment:
@gagan-aryan this looks like an error : you're using the same isotope value (iso1) for all partition functions ?